lib/inertia_rails/renderer.rb

```diff
@@ -112,5 +115,32 @@ def resolve_component(component)
 
       configuration.component_path_resolver(path: controller.controller_path, action: controller.action_name)
     end
+
+    def validate_partial_props(merged_props)
+      return unless rendering_partial_component?
+
+      unrequested_props = merged_props.select do |key, prop|
+        !key.in?(partial_keys)
+      end
+
+      props_that_will_unecessarily_compute = unrequested_props.select do |key, prop|
+        !prop.respond_to?(:call)
+      end
+
+      if props_that_will_unecessarily_compute.present?
+        props = props_that_will_unecessarily_compute.keys.map { |k| ":#{k}" }.join(', ')
+        is_plural = props_that_will_unecessarily_compute.length > 1
```

Comment:
I'd say it's too much for an error message 😅

Reply:
haha fair enough